### ISSUE TITLE
Add method to immediately leave operation successfully 

### ISSUE THREAD

**Issue by dnd** [trailblazer/trailblazer-operation#6]
It is easy to immediately leave an operation with a failed status by simply returning a falsey value, or raising an exception and ending up on the left track. There is no option to immediately leave(short circuit) an operation with a successful status. The only option currently is to place a value into context, and check it in every remaining step. 

Maybe there could be a method along the lines of `complete!` or `done!` that can be called from within a step to throw a symbol and immediately send the operation to the end, but still retain a successful result. 

Another option would be to take a path similar to Representable and return a specific constant that will be checked for(eg `Representable::Pipeline::Stop`).

**Comment by pvmeerbe** [comment]
I'm using 

    return Railway.pass_fast!

inside my operation step for this case, which seems to work fine

**Comment by dnd** [comment]
Yes, this was added in a month ago or so. At this point though @apotonick
does not want to document it, as he feels it trends more towards a full
workflow implementation. So whether this will stay permanently or not is
unknown.

**Comment by apotonick** [comment]
The 2.0.3 API will stay, @dnd, however, I've been working on advanced workflow semantic backed by BPMN diagrams. They will be the way to go when having a non-linear flow. The good news is, they integrate seamlessly with the operation. So, instead of trying to press everything into the Railway, I've introduced a new layer. You will love it. :stuck_out_tongue_closed_eyes:

**Comment by dikond** [comment]
I would love to use such feature in our operations! I was about to propose the same, but found this issue : )

We use `dry-matcher` gem and it isn't a problem for us to deviate to the left track. If we set a flag in the `options` object that signalizes that some steps were intentionally skipped, we can treat such operations as `successful`. Although they are _technically_ failed...

**Comment by apotonick** [comment]
In 2.1, you can have any workflow in the operation, and different outcomes. It will be released shortly.

Can you show us how that pattern matching looks in the operation, @dikond ? That'd be amazing.

**Comment by dikond** [comment]
We are doing matching in the controller actions (Hanami). We've created the `endpoint` helper that performs that. The whole setup looks like:

The helper:
```ruby
# included in all actions
module Shared
  module Endpoint
    Matcher = Dry::Matcher.new(
      success: Dry::Matcher::Case.new(
        match:   ->(result) { result.success? || result[SKIPPED_FLAG] },
        resolve: ->(result) { result }
      ),
      failure: Dry::Matcher::Case.new(
        match:   ->(result) { result.failure? },
        resolve: ->(result) { result }
      )
    )

    def endpoint(operation, opts = {}, &block)
      result = operation.(params.to_h, opts)
      Matcher.(result, &block)
    end
  end
end
```

The action:
```ruby
module Web::Controllers::Things
  class Create
    include Web::Action
    include App::Import[create_thing: 'things.operations.create']

    def call(_params)
      authorize! :things, :create

      endpoint(create_thing) do |m|
        m.success do
          redirect_to routes.settings_path(:things)
        end

        m.failure do |result|
          # handle errors
        end
      end
    end
  end
end
```

The operation:
```ruby
module App::Things
  module Operations
    class Create < Core::Operation
      schema Things::Validations::Form

      step Validate()
      step :check_if_thing_changed!
      step :create_thing!

      def check_if_thing_changed!(opts)
        # ... some code here check validated params against what we have now in db
        return true if changed?

        # we deviate to the left track here, operation failed
        opts[SKIPPED_FLAG] = true
        false
      end

      # this won't be called
      def create_thing!(opts)
        # ...
      end
    end
  end
end
```
Currently, we use the custom macro for skipping, so no need to set this flag manually. I've tried to show the general idea.

**Comment by apotonick** [comment]
Ok this looks very close to our planned `endpoint` gem. Thanks!

**Comment by apotonick** [comment]
Hey @dikond great that we met last weekend. Can we discuss the new 2.1 API and your requirements here? It's very important to get some input before pushing the new version. :beers:

**Comment by panSarin** [comment]
@dikond 3 years passed, 2.1 was released, I believe that issue is outdated. And please join us at zulipchat https://trailblazer.zulipchat.com/join/od4yhihn6tra3ix39u4gks72/  .

**Comment by apotonick** [comment]
This is implemented in 2.1 via the new wiring API and the `End()` helper: https://2019.trailblazer.to/2.1/docs/activity.html#activity-wiring-api-end- :beers:
